sdk/storage/Azure.Storage.Common/src/Shared/TaskExtensions.cs

```diff
@@ -1,110 +0,0 @@
-﻿// Copyright (c) Microsoft Corporation. All rights reserved.
-// Licensed under the MIT License.
-
-using System;
-using System.Diagnostics;
-using System.Threading.Tasks;
-
-namespace Azure.Storage
-{
-    /// <summary>
-    /// Extensions to ensure async Tasks execute synchronously.
-    /// </summary>
-    internal static class TaskExtensions
-    {
-        /// <summary>
-        /// Ensure the Task has finished executing.
-        /// </summary>
-        /// <typeparam name="T">Task's return type.</typeparam>
-        /// <param name="task">The task.</param>
-        /// <returns>The result of executing the task.</returns>
-        public static T EnsureCompleted<T>(this Task<T> task)
-        {
-            VerifyTaskCompleted(task);
```

Comment:
This isn't the same as the `TaskExtensions` you're replacing it with.  We're losing the validation.

Reply:
Both this class and `Azure.Core/src/Shared/TaskExtensions.cs` have validation, but this class throws `InvalidOperationException` to avoid NUnit issues with `Debug.Assert()`. @pakrym , do you think we should use `VerifyTaskCompleted` in shared code?

Reply:
Added `VerifyTaskCompleted` to `Azure.Core`.